plugins/module_utils/job.py

```diff
@@ -122,430 +114,190 @@ def _job_not_found(job_id, owner, job_name, dd_name, ovrr=None):
-    match = re.search(
-        r"(?:\s*CC\s*([0-9]+))|(?:ABEND\s*((?:S|U)[0-9]+)|(?:JCL ERROR))", rc_str
-    )
-    if match:
-        rc = match.group(1) or match.group(2)
-    return rc
+def _zget_job_status(job_id="*", owner="*", job_name="*", dd_name=None):
+    if job_id == "*":
+        job_query = None
+    else:
+        job_query = job_id
+
+    # entries = listing(job_query, owner)   1.2.0 has owner paramn, 1.1 does not
+    entries = listing(job_query)
+
+    final_entries = []
+    if entries:
+        for entry in entries:
+            if owner != "*":
+                if owner != entry.owner:
+                    continue
+            if job_name != "*":
+                if job_name != entry.name:
+                    continue
+
+            job = {}
+
+            job["job_id"] = entry.id
+            job["job_name"] = entry.name
+            job["subsystem"] = ""
+            job["system"] = ""
+            job["owner"] = entry.owner
+
+            job["ret_code"] = {}
+            job["ret_code"]["msg"] = entry.status + " " + entry.rc
+            job["ret_code"]["msg_code"] = entry.rc
+
+            job["ret_code"]["code"] = ""
+            if len(entry.rc) > 0:
+                if entry.rc.isdigit():
+                    job["ret_code"]["code"] = int(entry.rc)
+
+            job["ret_code"]["msg_text"] = entry.status
+
+            job["class"] = ""
+            job["content_type"] = ""
+
+            job["ret_code"]["steps"] = []
+            job["ddnames"] = []
+            list_of_dds = list_dds(entry.id)
+
+            for single_dd in list_of_dds:
+                dd = {}
+                if dd_name is not None:
+                    if dd_name not in single_dd["dataset"]:
+                        continue
+
+                if "dataset" not in single_dd:
+                    continue
+
+                dd["ddname"] = single_dd["dataset"]
+                if "recnum" in single_dd:
+                    dd["record_count"] = single_dd["recnum"]
+                else:
+                    dd["record_count"] = "n/a"
+
+                if "dsid" in single_dd:
+                    dd["id"] = single_dd["dsid"]
+                else:
+                    dd["id"] = "?"
+
+                if "stepname" in single_dd:
+                    dd["stepname"] = single_dd["stepname"]
+                else:
+                    dd["stepname"] = "UNAVAIL"
+
+                if "procstep" in single_dd:
+                    dd["procstep"] = single_dd["procstep"]
+                else:
+                    dd["proctep"] = None
+
+                if "length" in single_dd:
+                    dd["byte_count"] = single_dd["length"]
+                else:
+                    dd["byte_count"] = 0
+
+                if ("stepname" in single_dd) and ("dataset" in single_dd):
+                    tmpcont = read_output(entry.id, single_dd["stepname"], single_dd["dataset"])
+                else:
+                    tmpcont = "no content available"
```

Comment:
Wouldn't it be more efficient if you set tmpcont to None and then in some of the following logical checks eg; `if len(job["class"]) < 1` could be `if tmpcontent and  len(job["class"]) < 1` or wrap the entire block in another `if tmpcontent`, not a deal breaker just a suggestion.

Reply:
I switched to nested if with a Non default.  I'm just starting to get into the 'avoid else' thing.